### PR TITLE
Adds retry instead of immediate failure

### PR DESCRIPTION
**Pull request by ScottGarman**
## Description

Instead of immediate failure, we now have a retry mechanism in the event DHCP is slow

## Why is this needed

Instant failure happens if the network takes a little bit of time to come up.. this adds a retry and sleep etc.

Fixes: #

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade

### CODE REVIEW

File: tink-docker/main.go [hunk 1]
```diff
@@ -79,11 +79,28 @@ func parseCmdLine(cmdLines []string) (cfg tinkConfig) {
...
+	timeOut = time.Millisecond * 500 // 0.5 seconds
+	var resp *http.Response
+	var err error
+
+	// Retry this task
+	for retries := 0; retries < maxRetryCount; retries++ {
 	// Get the data
-	resp, err := http.Get(url)
-	if err != nil {
+	resp, err = http.Get(url)
```
Comment (stephen-fox) [inline tink-docker/main.go:95]: Could you try running `go fmt` on this file? It looks like the formatting had a bad time.

Comment (ScottGarman) [reply tink-docker/main.go]: Done.

Comment (stephen-fox) [reply tink-docker/main.go]: Thank you :)

File: tink-docker/main.go [hunk 2]
```diff
@@ -79,10 +79,27 @@ func parseCmdLine(cmdLines []string) (cfg tinkConfig) {
...
+	var timeOut time.Duration
+	maxRetryCount = 10
+	timeOut = time.Millisecond * 500 // 0.5 seconds
+	var resp *http.Response
+	var err error
+
+	// Retry this task
+	for retries := 0; retries < maxRetryCount; retries++ {
+		// Get the data
+		resp, err = http.Get(url)
```
[the remark this answers is not in the document]
Comment (ScottGarman) [reply tink-docker/main.go:95]: After some discussion, we feel that the caller of downloadFile() has the responsibility to wrap the call in a timeout function if needed.

**Review by stephen-fox — APPROVED** [review]
We discussed the HTTP GET timeout as a team, and decided that the caller can deal with it because Go has an awesome concurrency model.

**Pull request merged**